clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -93,81 +135,114 @@ func (a *apiAvailabilityMeasurement) start(config *measurement.Config, SSHToMast
 			select {
 			case <-a.stopCh:
 				return
-			case <-time.After(time.Duration(probeDuration)):
+			case <-a.holdCh:
+				select {
+				case <-a.stopCh:
+					return
+				case <-a.holdCh:
+					continue
+				}
+			case <-time.After(a.frequency):
 				a.updateClusterAvailabilityMetrics(k8sClient)
-				if SSHToMasterSupported {
-					a.updateMasterAvailabilityMetrics(k8sClient, config, provider)
+				if a.masterLevelAvailabilityEnabled() {
```

Comment:
would checking `config.ClusterLoaderConfig.EnableExecService` be more appropriate here? `masterLevelAvailabilityEnabled` is about whether master level metrics have been collected and should probably be renamed

Reply:
Apart from exec service being enabled we also require the internal IPs of masters for the per-VM availability, so it wouldn't suffice. `masterLevelAvailabilityEnabled` boolean function is true iff we are running per-VM polling in the measurement as we fill the `a.hostIPs` field after we check that both exec service is enabled and the internal IPs are provided.